### PR TITLE
[MON-21] Add functionality to modify email recipients

### PR DESCRIPTION
**Pull request by sri-9597**

### CODE REVIEW

File: monitor/src/main/java/edu/asu/diging/monitor/core/db/impl/NotificationRecipientDbConnection.java [hunk 1]
```diff
@@ -42,6 +42,16 @@ public INotificationRecipient store(INotificationRecipient recipient) throws Uns
         return recipient;
     }
 	
+	@Override
+	public INotificationRecipient update(INotificationRecipient recipient) throws UnstorableObjectException {
+	    if (recipient.getEmail() == null) {
+            throw new UnstorableObjectException("Recipient doesn't have an email address.");
+        }
```
Comment (jdamerow) [inline monitor/src/main/java/edu/asu/diging/monitor/core/db/impl/NotificationRecipientDbConnection.java:49]: indentation

File: monitor/src/main/java/edu/asu/diging/monitor/core/exceptions/NoEmailRecipientException.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package edu.asu.diging.monitor.core.exceptions;
+
+public class NoEmailRecipientException extends Exception {
+
+    private static final long serialVersionUID = -3658335972369001233L;
```
Comment (jdamerow) [inline monitor/src/main/java/edu/asu/diging/monitor/core/exceptions/NoEmailRecipientException.java:5]: should start at 1

File: monitor/src/main/java/edu/asu/diging/monitor/core/service/INotificationManager.java [hunk 3]
```diff
@@ -1,22 +1,25 @@
...
+	boolean modifyRecipient(String email, String name, List<String> apps) throws NoEmailRecipientException;
+
+	void sendNotificationEmails(IApp app, AppStatus previousStatus);
+	
+	List<INotificationRecipient> getAllRecipients();
+	
+	void deleteRecipient(String email);
+	
+	INotificationRecipient getRecipient(String id);
+
```
Comment (jdamerow) [inline monitor/src/main/java/edu/asu/diging/monitor/core/service/INotificationManager.java:24]: looks like you're changing line ending here

File: monitor/src/main/java/edu/asu/diging/monitor/core/service/impl/NotificationManager.java [hunk 4]
```diff
@@ -1,97 +1,123 @@
...
+            recipients = Arrays.asList(dbConnection.getAllRecipients());
+        } else {
+            recipients = app.getRecipients();
+        }
+        for (INotificationRecipient recipient : recipients) {
+            emailManager.sendAppStatusNotificationEmail(recipient.getEmail(), app, previousStatus.toString(),
+                    app.getLastAppTest().getStatus().toString());
+        }
+    }
+}
```
Comment (jdamerow) [inline monitor/src/main/java/edu/asu/diging/monitor/core/service/impl/NotificationManager.java:123]: same here

**Review by jdamerow — CHANGES_REQUESTED** [review]
Please fix your line endings. I'll continue reviewing after. Please look at your PRs after/before you create them, I should not have to make these kind of comments, since it should be obvious to see when you look at the changes.

File: monitor/src/main/java/edu/asu/diging/monitor/core/exceptions/NoEmailRecipientException.java (hunk 2 above)
Comment (sri-9597) [reply monitor/src/main/java/edu/asu/diging/monitor/core/exceptions/NoEmailRecipientException.java:5]: Sorry Julia, I'm not sure I understand. Is this regarding the serialVersionUID?

Comment (jdamerow) [reply monitor/src/main/java/edu/asu/diging/monitor/core/exceptions/NoEmailRecipientException.java:5]: Yes

**Pull request closed without merging**